jwst/rscd/rscd_sub.py

```diff
@@ -135,58 +250,136 @@ def get_rscd_parameters(input_model, rscd_model):
-    scale2_odd = scale2_table[index2]
-
-    even = tau1_even, scale1_even, tau2_even, scale2_even
-    odd = tau1_odd, scale1_odd, tau2_odd, scale2_odd
-
-    return even, odd
+    param['odd']['tau'] = tau_table[index2]
+    param['odd']['ascale'] = ascale_table[index2]
+    param['odd']['pow']  = pow_table[index2]
+    param['odd']['illum_zp'] = illum_zp_table[index2]
+    param['odd']['illum_slope'] = illum_slope_table[index2]
+    param['odd']['illum2'] = illum2_table[index2]
+    param['odd']['param3'] = param3_table[index2]
+    param['odd']['crossopt'] = crossopt_table[index2]
+    param['odd']['sat_zp'] = sat_zp_table[index2]
+    param['odd']['sat_slope'] = sat_slope_table[index2]
+    param['odd']['sat2'] = sat_2_table[index2]
+    param['odd']['sat_mzp'] = sat_mzp_table[index2]
+    param['odd']['sat_rowterm'] = sat_rowterm_table[index2]
+    param['odd']['sat_scale'] = sat_scale_table[index2]
+
+    return param
 
 
 def get_DNaccumulated_last_int(input_model, i, sci_ngroups):
 
-    # Find the accumulated DN from the last integration
-    # need to add skipping N frames (frames affected by reset)
-    # Add check to make sure we have enough frames left to do a fit
+    """
+    Find the accumulated DN from the last integration
+    This data should already have the Reset Anomaly correction
+    applied (if not - should we skip frames at the beginning ?) 
+
+    a Check has already been made to make sure we have at least 
+    4 frames 
 
+    Parameters
+    ----------
+    input_model: input ramp data
+    i: integration #
+    sci_ngroups: number of frames/integration
+    
+    return values
+    -------------
+    sat: is the previous integration for this pixel saturated: yes/no
+    is_ref: boolean if pixel is a reference pixel. Do not apply correction to these pixels
+            we set lastframe = 0 for these pixels 
+    dn_lastframe_23: the extrapolated last frame using second and third to last frames
+    dn_lastfrane_fit: the extrapolated last frame using the fit to the entire ramp
+   
+    """
+ 
     nrows = input_model.data.shape[2]
     ncols = input_model.data.shape[3]
 
-    # last frame affected by "last frame" effect - use second to last frame
-    # may want to extrapolate to last frame
-    # we may want to check if data has saturated
-    dn_lastframe = input_model.data[i - 1][sci_ngroups - 2]
-
-    dn_accumulated = dn_lastframe.copy() * 0.0
+    dn_lastframe2 = input_model.data[i - 1][sci_ngroups - 2]
+    dn_lastframe3 = input_model.data[i - 1][sci_ngroups - 3]
+    dn_lastframe23 = dn_lastframe2.copy() * 0.0
+    dn_lastframe_fit = dn_lastframe2.copy() * 0.0
+    saturated = np.full((nrows,ncols),False)
+    is_ref = saturated.copy()
+
+    diff = dn_lastframe2 - dn_lastframe3
+    dn_lastframe23 = dn_lastframe2 + diff
+    # check if pixel is saturated
+    saturated_flag = datamodels.dqflags.group['SATURATED']
+    ref_flag = datamodels.dqflags.pixel['REFERENCE_PIXEL']
+    
     for j in range(nrows):
```

Comment:
Someday it would be nice to find a way to do this without the looping over pixels, which can be really slow in python. Especially because it's being done for every integration. But if it works as-is, without being ridiculously slow, let's leave it for the future.